src/algebra/monoid_algebra/basic.lean

```diff
@@ -1099,6 +1103,15 @@ lemma of'_eq_of [add_zero_class G] (a : G) : of' k G a = of k G a := rfl
 lemma of_injective [nontrivial k] [add_zero_class G] : function.injective (of k G) :=
 λ a b h, by simpa using (single_eq_single_iff _ _ _ _).mp h
 
+/--
+Embedding of the product of a `semiring k` and a `add_zero_class G` into the
+`add_monoid_algebra k G`
+-/
+@[simps] def monomial_hom [add_zero_class G] : k × multiplicative G →* add_monoid_algebra k G :=
+{ to_fun := λ a , single a.2 a.1,
```

Comment:
```suggestion
{ to_fun := λ a, single a.2.to_add a.1,
```
which makes it clear where you're unpacking `multiplicative`, and ensures the resulting `simp` lemma is usable.